serverless/aws/cdk/src/main/scala/sttp/tapir/serverless/aws/cdk/core/Url.scala

```diff
@@ -0,0 +1,33 @@
+package sttp.tapir.serverless.aws.cdk.core
+
+import sttp.tapir.internal.UrlencodedData
+
+sealed trait Segment {
```

Comment:
The naming convention for files would be that if there's a main class/interface, then the name of the file follows that name (capitalized). Here the file should be `Segment.scala`.

If there are multiple data structures which don't have a strong "dominating" component, than the file is lower-cased, e.g. `url.scala`. But here I'd stick with segment.

Also something that's not really a convention, as people tend to write using different styles, but often sealed trait implementations are neste din the companion object, keeping the namespace claner, so you'd have `Segment.Fixed` etc. This in fact follows how Scala 3's enums work.

Reply:
Thank you, that is really valuable remark for me